tests/python/pants_test/pants_run_integration_test.py

```diff
@@ -177,6 +177,8 @@ def hermetic_env_whitelist(cls):
     return [
         # Used in the wrapper script to locate a rust install.
         'HOME',
+        # Needed to find python interpreters and other binaries. 
+        'PATH',
```

Comment:
This is a really big leak in hermeticity.

If you're able to get this working by adding only our current interpreter as the PATH: https://github.com/pantsbuild/pants/blob/4de08d613d9e70a53951b3ca2869835af6fd7b32/src/python/pants/backend/python/rules/python_test_runner.py#L68 ... that would be preferable.

Reply:
Alas this does not work for tests that expect to select a different interpreter. Not to mention tests that require a JVM or whatever. 

Note that PATH was always leaky even in hermetic integration tests: When we don't explicitly set it (which we almost never do), the spawned pants process will get the default PATH for the shell (e.g., as specified in `/etc/profile`). We just never notice this because that PATH happens to work for us in practice. At least this way the PATH setting is explicit. 

Having every test control its path is probably desirable, but that's a major piece of work that I don't want to bite off right now...

Reply:
Also note that even setting PATH to just the dirs containing binaries we know we need is not particularly hermetic, since we don't know what else is in those dirs. The PATH mechanism is intrinsically non-hermetic, unless we control the directory contents, say via a symlink farm.

Reply:
> Note that PATH was always leaky even in hermetic integration tests: When we don't explicitly set it (which we almost never do), the spawned pants process will get the default PATH for the shell (e.g., as specified in /etc/profile). We just never notice this because that PATH happens to work for us in practice. At least this way the PATH setting is explicit.

While this is certainly true, I remain suspicious of opening the door to a _completely_ unfiltered PATH here. The only thing that should have changed here is the use of a `pants.pex` rather than a `pants` script, so our method of finding other tools should not have been affected in any way.

Reply:
TBH I no longer remember why this was necessary, but I did at one point. I'll remove this and see what breaks, maybe that will jog my memory.

That said, I still think I prefer an unfiltered but explicit PATH passthru, so that which path is used is clear to the reader, rather than what we have today, which is some random implicit PATH that only happens to work by chance.

Reply:
If you want to land the rest of the improvements here and follow up as a TODO, that would be fine with me. It's unlikely for something to slip through this hole in the next week.

Reply:
IIRC, at least one issue is pexes use a shebang of `#!/usr/bin/env ...` by default. Explicitly setting the shebang to a full binary path may be one of the fixes needed if anyone does circle back to this.